### DISCUSSION TITLE
Help on trying to install and build valhalla on my mac

### DISCUSSION THREAD

**Discussion by fiotbl** [valhalla/valhalla#3623] [Q&A]
Hi guys, I have been trying to get Valhalla running on my mac but I am quite confused with the documentation especially on running on docker and building (for my mac). Just wondering if anyone can guide me through step by step on how to get it running, please? Thank you so much

**Comment by nilsnolde** [comment]
That’s actually the README‘s job. If you let us know what you’re trying and what the errors are, we can support better, and 
 patch the README.

**Reply by fiotbl** [reply]
Hi, after I have checkouted your run-* docker containers here: https://hub.docker.com/r/valhalla/valhalla/tags using 
`docker pull valhalla/valhalla:run-latest `
I am now trying to run the bash script under the "Running" section. However, when I try to run the 
`valhalla_build_config --mjolnir-tile-dir ${PWD}/valhalla_tiles --mjolnir-tile-extract ${PWD}/valhalla_tiles.tar --mjolnir-timezone ${PWD}/valhalla_tiles/timezones.sqlite --mjolnir-admin ${PWD}/valhalla_tiles/admins.sqlite > valhalla.json
`
I get this error `zsh: command not found: valhalla_build_config`

**Reply by fiotbl** [reply]
Am I missing any steps?

**Comment by nilsnolde** [comment]
Either you run docker (and then perform all things from within the docker container), or you build Valhalla for your Mac natively, ie follow all the other steps in the README before running valhalla_build_config.

**Reply by fiotbl** [reply]
I am trying to run docker using this image https://github.com/gis-ops/docker-valhalla but I am getting this error after trying to run the docker:

```
INFO: Running container with user valhalla UID 59999 and GID 59999.


WARNING: Hash not found for: /custom_files/andorra-latest.osm.pbf!

WARNING: No routing tiles found, starting new build..

INFO: Running build tiles with: /custom_files/valhalla.json /custom_files/andorra-latest.osm.pbf


============================

= Build the initial graph. =

============================

2022/05/11 08:12:39.685900 [INFO] Start stage = initialize End stage = build

qemu: uncaught target signal 11 (Segmentation fault) - core dumped

/valhalla/scripts/configure_valhalla.sh: line 172:    91 Segmentation fault      valhalla_build_tiles -c ${CONFIG_FILE} -e build ${files}
```

**Reply by nilsnolde** [reply]
Hm that’s strange.. What was the exact command you did? I’ll try to have a look myself later today

**Reply by nilsnolde** [reply]
Btw, this is not really the right place for that issue, this would be: https://github.com/gis-ops/docker-valhalla. I’ll open an issue there when I can confirm

**Reply by fiotbl** [reply]
Okay sure! 
Now i am trying by running `docker run -dt --platform linux/amd64 --name valhalla -p 8002:8002 -v /MY_PATH/custom_files:/custom_files valhalla/valhalla:run-latest` after checking out from https://hub.docker.com/r/valhalla/valhalla/tags directly. 
The container seems to be running, but after running `curl http://localhost:8002/route --data '{"locations":[{"lat":47.365109,"lon":8.546824,"type":"break","city":"Zürich","state":"Altstadt"},{"lat":47.108878,"lon":8.394801,"type":"break","city":"6037 Root","state":"Untere Waldstrasse"}],"costing":"auto","directions_options":{"units":"miles"}}' | jq '.'` I get the error `curl: (52) Empty reply from server`

I have already downloaded the OSM data into my customer_files folder

**Reply by nilsnolde** [reply]
Those two images are _completely_ different as to what they do under the hood. The valhalla native one is actually just an artifacts dump. You can use it sure, but not in the way you’re trying, everything has to be done _inside_ the container. That’s the exact reason why we came up with our own docker image, which has many automations to make your life easier

**Reply by fiotbl** [reply]
So I should stick to using the image from https://github.com/gis-ops/docker-valhalla?

**Reply by brad-4k** [reply]
@fiotbl I ran the image from https://github.com/gis-ops/docker-valhalla with very little effort.  I just followed the instructions in the README and it worked like magic.
